public/app/plugins/datasource/prometheus/querybuilder/components/MetricsLabelsSection.tsx

```diff
@@ -68,12 +68,7 @@ export function MetricsLabelsSection({
     labelsToConsider.push({ label: '__name__', op: '=', value: query.metric });
     const expr = promQueryModeller.renderLabels(labelsToConsider);
 
-    let labelsIndex: Record<string, string[]>;
-    if (datasource.hasLabelsMatchAPISupport()) {
-      labelsIndex = await datasource.languageProvider.fetchSeriesLabelsMatch(expr);
-    } else {
-      labelsIndex = await datasource.languageProvider.fetchSeriesLabels(expr);
-    }
+    let labelsIndex: Record<string, string[]> = await datasource.languageProvider.fetchLabelsWithMatch(expr);
```

Comment:
removing the check to `hasLabelsMatchAPISupport` will force all prometheus to use the new endpoints, even if they don't support them. This would cause users with old prometheus/thanos/cortex versions to return invalid results.

Probably something we want to do eventually, but maybe not in this PR?

Reply:
I didn't remove the `hasLabelsMatchAPISupport ` check, I moved it. 
The check now is inside `fetchLabelsWithMatch()` in one place rather than distributed across the code.
So, nothing is changed for datasources with `hasLabelsMatchAPISupport == false`.
Just better code organization.

Reply:
@gtk-grafana Could you please reconsider `breaking change` label? I don't think it is applicable to this PR